modules/imgproc/include/opencv2/imgproc.hpp

```diff
@@ -261,6 +261,8 @@ enum InterpolationFlags{
     /** flag, fills all of the destination image pixels. If some of them correspond to outliers in the
     source image, they are set to zero */
     WARP_FILL_OUTLIERS   = 8,
+    /** PIL nearest neighbor interpolation */
+    INTER_NEAREST_PIL = 9,
```

Comment:
It would be nice to have slightly more verbose description in the documentation, something like `differs from INTER_NEAREST in the way it calculates pixel coordinates: center of the square instead of top left corner. For example, pixel (0, 0) will have coordinates (0.5, 0.5) in this mode.`

Reply:
Won't this new enum value interfere with flags? `INTER_NEAREST_PIL = 9 // 1001` will imply `WARP_FILL_OUTLIERS = 8 // 1000` flag set. Maybe it is better to add new flag: `WARP_SQUARE_PIXELS` or `WRAP_SHIFT_COORDINATES` with value `32`?